### PR TITLE
doc/website/blog: v0.15 announcement

### PR DESCRIPTION
**Pull request by rotemtam**

### CODE REVIEW

File: doc/website/blog/2023-10-19-atlas-v-0-15.mdx [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+
+Let's dive right in!
+
+### Interactive Declarative Migrations
+
+Atlas supports a Terraform-like workflow for managing your database schema using the `schema apply` command.
+This workflow, which we call "Declarative Migrations", is a modern alternative to the traditional "versioned migrations"
+workflow. In declarative migrations, you define your desired schema in one of the formats supported by Atlas and supply
+a connection string to your database. Atlas compares the current and desired schema of your database and generates a
+migration plan to bring your database to the desired state.
```
Comment (masseelch) [inline doc/website/blog/2023-10-19-atlas-v-0-15.mdx:36]:
```suggestion
plan to migrate your database to the desired state.
```

File: doc/website/blog/2023-10-19-atlas-v-0-15.mdx [hunk 2]
```diff
@@ -0,0 +1,459 @@
...
+1. *Ensuring Safety* - you can count on Atlas to generate a *correct* migration plan to your desired state,
+   but it's still possible that this migration will have unintended side effects. For example, adding a `UNIQUE`
+   constraint to a column might fail if there are duplicate values in the column.
+2. *Editing* - users often want to make changes to their migration plan before applying it. In the current flow, this
+   requires running `schema apply` with the `--dry-run` flag, saving the output to a file, editing it, and then manually
+   applying the edited migration plan to the database.
+
+#### Enter: Interactive Declarative Migrations
+
+In this release, we are introducing a new "Lint and Edit" mode to the `schema apply` command. This mode is
```
[the remark this answers is not in the document]
Comment (masseelch) [reply doc/website/blog/2023-10-19-atlas-v-0-15.mdx:51]: Feels repetitive and reading flow disruptive the way it is currently after the new heading.

File: doc/website/blog/2023-10-19-atlas-v-0-15.mdx [hunk 3]
```diff
@@ -0,0 +1,459 @@
...
+available to logged-in users only, as it uses Atlas Cloud to provide a neat UI and rich analysis capabilities.
+Let's see it in action.
+
+Start by downloading the latest version of Atlas:
+
+```
+curl -sSf https://atlasgo.sh | sh
+```
+For installation instructions on other platforms, see the [installation guide](https://atlasgo.io/getting-started#installation).
```
Comment (masseelch) [inline doc/website/blog/2023-10-19-atlas-v-0-15.mdx:60]:
```suggestion
For more in-depth installation instructions, see the [installation guide](https://atlasgo.io/getting-started#installation).
```

File: doc/website/blog/2023-10-19-atlas-v-0-15.mdx [hunk 4]
```diff
@@ -0,0 +1,459 @@
+---
+title: "Announcing v0.15: Interactive Declarative Migrations, Functions, and Stored Procedures"
+authors: rotemtam
+tags: [announcement, atlas, declarative, functions, stored-procedures]
```
Comment (a8m) [inline doc/website/blog/2023-10-19-atlas-v-0-15.mdx:4]:
```suggestion
title: "Announcing v0.15: Interactive Declarative Migrations, Functions, Procedures and Domains"
authors: rotemtam
tags: [announcement, atlas, declarative, functions, stored-procedures]
```

File: doc/website/blog/2023-10-19-atlas-v-0-15.mdx [hunk 5]
```diff
@@ -0,0 +1,459 @@
...
+    type = varchar(100)
+    default = "Anonymous"
+  }
+}
+```
+
+Now, let's apply this schema to a local SQLite database named "sqlite.db":
+
+```
+atlas schema apply -u sqlite://sqlite.db -f schema.hcl
```
Comment (masseelch) [inline doc/website/blog/2023-10-19-atlas-v-0-15.mdx:86]:
```suggestion
atlas schema apply -u "sqlite://sqlite.db" -f schema.hcl
```

File: doc/website/blog/2023-10-19-atlas-v-0-15.mdx [hunk 6]
```diff
@@ -0,0 +1,459 @@
...
+tags: [announcement, atlas, declarative, functions, stored-procedures]
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+Hi everyone!
+
+It's been a few weeks since our last version announcement and today I'm happy to share with you  
+[v0.15](https://github.com/ariga/atlas/releases/tag/v0.14.0), which includes some
```
Comment (a8m) [inline doc/website/blog/2023-10-19-atlas-v-0-15.mdx:13]:
```suggestion
[v0.15](https://github.com/ariga/atlas/releases/tag/v0.15.0), which includes some
```

File: doc/website/blog/2023-10-19-atlas-v-0-15.mdx [hunk 7]
```diff
@@ -0,0 +1,459 @@
...
+
+Notice the new "Lint and edit" option. Select it and press Enter. Atlas will now analyze the migration plan and
+open your browser in the new, interactive migration plan screen. Notice the screen contains three important sections:
+
+* **Migration Plan** - the migration plan generated by Atlas. You can click the "Edit" button to make changes to it.
+* **Checks** - a summary of the checks that Atlas ran against the generated plan. In this case, our plan is completely
+safe, so all checks passed.
+* **ERD** - A visual representation of the change we are planning.
+
+![](https://atlasgo.io/uploads/blog/v0.15/interactive-plan.png)
```
Comment (masseelch) [inline doc/website/blog/2023-10-19-atlas-v-0-15.mdx:111]: Does not exist

File: doc/website/blog/2023-10-19-atlas-v-0-15.mdx (hunk 1 above)
Comment (a8m) [reply doc/website/blog/2023-10-19-atlas-v-0-15.mdx:36]: Rotem's version feels more clear to me.

File: doc/website/blog/2023-10-19-atlas-v-0-15.mdx [hunk 8]
```diff
@@ -0,0 +1,459 @@
...
+![](https://atlasgo.io/uploads/blog/v0.15/interactive-destructive.png)
+
+In this case, we decide to abort the migration in order to not lose the precious data on the
+`hello` table. Good thing we have automatic migration linting on our side!
+
+### Functions and Stored Procedures
+
+Over the past few months, we have received numerous requests to support management of [Functions](https://www.postgresql.org/docs/current/functions.html)
+and [stored procedures](https://www.postgresql.org/docs/current/sql-createprocedure.html) in popular databases such as
+[PostgreSQL](https://postgresql.org) and [MySQL](https://mysql.com).  Functions and stored procedures are a way to encapsulate
```
Comment (masseelch) [inline doc/website/blog/2023-10-19-atlas-v-0-15.mdx:165]:
```suggestion
[PostgreSQL](https://postgresql.org) and [MySQL](https://mysql.com). Functions and stored procedures are a way to encapsulate
```

File: doc/website/blog/2023-10-19-atlas-v-0-15.mdx (hunk 6 above)
Comment (giautm) [inline doc/website/blog/2023-10-19-atlas-v-0-15.mdx:13]:
```suggestion
[v0.15](https://github.com/ariga/atlas/releases/tag/v0.15.0), which includes some
```

File: doc/website/blog/2023-10-19-atlas-v-0-15.mdx [hunk 9]
```diff
@@ -0,0 +1,459 @@
...
+open your browser in the new, interactive migration plan screen. Notice the screen contains three important sections:
+
+* **Migration Plan** - the migration plan generated by Atlas. You can click the "Edit" button to make changes to it.
+* **Checks** - a summary of the checks that Atlas ran against the generated plan. In this case, our plan is completely
+safe, so all checks passed.
+* **ERD** - A visual representation of the change we are planning.
+
+![](https://atlasgo.io/uploads/blog/v0.15/interactive-plan.png)
+
+![](https://atlasgo.io/uploads/blog/v0.15/interactive-erd.png)
```
Comment (a8m) [inline doc/website/blog/2023-10-19-atlas-v-0-15.mdx:113]: The links in these images are wrong.

File: doc/website/blog/2023-10-19-atlas-v-0-15.mdx [hunk 10]
```diff
@@ -0,0 +1,459 @@
...
+
+Hi everyone!
+
+It's been a few weeks since our last version announcement and today I'm happy to share with you  
+[v0.15](https://github.com/ariga/atlas/releases/tag/v0.14.0), which includes some
+very exciting improvements for Atlas:
+
+* **Interactive Declarative Migrations** - Atlas supports a Terraform-like workflow for managing your database schema
+  using the `schema apply` command. In this release we have added a new "Lint and Edit" mode to this command, which
+  will analyze your schema changes for issues and allow you to edit them interactively before applying them to your
```
Comment (hilakashai) [inline doc/website/blog/2023-10-19-atlas-v-0-15.mdx:18]:
```suggestion
  will analyze your schema changes for issues and will allow you to edit them interactively before applying them to your
```

File: doc/website/blog/2023-10-19-atlas-v-0-15.mdx [hunk 11]
```diff
@@ -0,0 +1,459 @@
...
+  database.
+* **Functions and Stored Procedures** - Atlas now supports creating and managing functions and stored procedures in
+  your database schema.
+* **Postgres Domains** - In addition, Atlas now supports [Postgres Domains](https://www.postgresql.org/docs/16/sql-createdomain.html)
+  as well. A domain is essentially a data type with optional constraints (restrictions on the allowed set of values).
```
Comment (hilakashai) [inline doc/website/blog/2023-10-19-atlas-v-0-15.mdx:23]:
```suggestion
. A domain is essentially a data type with optional constraints (restrictions on the allowed set of values).
```

**Review by masseelch — COMMENTED** [review]
:rock:

**Review by a8m — COMMENTED** [review]
🚀

File: doc/website/blog/2023-10-19-atlas-v-0-15.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (hilakashai) [reply doc/website/blog/2023-10-19-atlas-v-0-15.mdx:51]: Disagree :/ I would keep this sentence

File: doc/website/blog/2023-10-19-atlas-v-0-15.mdx [hunk 12]
```diff
@@ -0,0 +1,459 @@
...
+Use the arrow keys to navigate: ↓ ↑ → ←
+? Are you sure?:
+    Apply
+// highlight-next-line
+  ▸ Lint and edit # <-- Brand new!
+    Abort
+```
+
+Notice the new "Lint and edit" option. Select it and press Enter. Atlas will now analyze the migration plan and
+open your browser in the new, interactive migration plan screen. Notice the screen contains three important sections:
```
Comment (hilakashai) [inline doc/website/blog/2023-10-19-atlas-v-0-15.mdx:104]:
```suggestion
open your browser in the new, interactive migration plan screen. The screen contains three important sections:
```

File: doc/website/blog/2023-10-19-atlas-v-0-15.mdx [hunk 13]
```diff
@@ -0,0 +1,459 @@
...
+![](https://atlasgo.io/uploads/blog/v0.15/interactive-plan.png)
+
+![](https://atlasgo.io/uploads/blog/v0.15/interactive-erd.png)
+
+Once we are content with the migration plan, let's go ahead and click the "Approve and Apply" button. Atlas will
+apply the migration plan to the database and scroll down to the execution logs section:
+
+![](https://atlasgo.io/uploads/blog/v0.15/interactive-logs.png)
+
+Let's now edit our desired state a bit to delete the `hello` table and add a new `users` table:
```
Comment (hilakashai) [inline doc/website/blog/2023-10-19-atlas-v-0-15.mdx:120]:
```suggestion
Let's edit our desired state a bit to delete the `hello` table and add a new `users` table:
```

File: doc/website/blog/2023-10-19-atlas-v-0-15.mdx [hunk 14]
```diff
@@ -0,0 +1,459 @@
...
+This time, Atlas will warn us that the migration plan is not safe:
+
+![](https://atlasgo.io/uploads/blog/v0.15/interactive-destructive.png)
+
+In this case, we decide to abort the migration in order to not lose the precious data on the
+`hello` table. Good thing we have automatic migration linting on our side!
+
+### Functions and Stored Procedures
+
+Over the past few months, we have received numerous requests to support management of [Functions](https://www.postgresql.org/docs/current/functions.html)
```
Comment (hilakashai) [inline doc/website/blog/2023-10-19-atlas-v-0-15.mdx:163]:
```suggestion
Over the past few months, we have received numerous requests to support management of [functions](https://www.postgresql.org/docs/current/functions.html)
```

File: doc/website/blog/2023-10-19-atlas-v-0-15.mdx [hunk 15]
```diff
@@ -0,0 +1,459 @@
...
+To learn more about functions and stored procedures in Atlas, check out the [documentation](https://atlasgo.io/atlas-schema/hcl#function).
+
+### Postgres Domains
+
+Another highly requested feature was support for [Postgres Domains](https://www.postgresql.org/docs/16/sql-createdomain.html).
+A domain is essentially a data type with optional constraints (restrictions on the allowed set of values). For example,
+you might want to define an `email_address` domain which would be a `varchar` column with a `CHECK` constraint to ensure
+that the value is a valid email address.
+
+Starting v0.15, Atlas can now manage domains in your database schema, as well as use them as types for table columns.
```
Comment (hilakashai) [inline doc/website/blog/2023-10-19-atlas-v-0-15.mdx:332]:
```suggestion
Starting with v0.15, Atlas can now manage domains in your database schema, as well as use them as types for table columns.
```

File: doc/website/blog/2023-10-19-atlas-v-0-15.mdx [hunk 16]
```diff
@@ -0,0 +1,459 @@
...
+```
+
+### Support for TypeORM
+
+[TypeORM](https://typeorm.io/) is a popular ORM for Node.js. In this release, we are happy to announce the TypeORM
+integration, which allows you to automatically generate your database schema from your TypeORM entities, create visualizations,
+and more.
+
+The [TypeORM Atlas Provider](https://github.com/ariga/atlas-provider-typeorm) is a Node.js module that can extract
+the desired schema of your database directly from your TypeORM entities. To use it, first install it:
```
Comment (hilakashai) [inline doc/website/blog/2023-10-19-atlas-v-0-15.mdx:416]:
```suggestion
the desired schema of your database directly from your TypeORM entities. To use it, first install:
```

File: doc/website/blog/2023-10-19-atlas-v-0-15.mdx (hunk 3 above)
Comment (rotemtam) [reply doc/website/blog/2023-10-19-atlas-v-0-15.mdx:60]: The point is to let people know that other platforms exist

File: doc/website/blog/2023-10-19-atlas-v-0-15.mdx [hunk 17]
```diff
@@ -0,0 +1,459 @@
...
+Use the arrow keys to navigate: ↓ ↑ → ←
+? Are you sure?:
+  ▸ Apply
+    Abort
+```
+
+After applying, let's re-run the `schema apply` command to make sure that the schema is up-to-date:
+
+```text
+atlas schema apply -u 'postgres://postgres:pass@localhost:5432/postgres?sslmode=disable' -f schema.hcl
```
Comment (a8m) [inline doc/website/blog/2023-10-19-atlas-v-0-15.mdx:401]:
```suggestion
atlas schema apply -u 'postgres://postgres:pass@localhost:5432/postgres?sslmode=disable' --to file://schema.hcl
```

File: doc/website/blog/2023-10-19-atlas-v-0-15.mdx [hunk 18]
```diff
@@ -0,0 +1,459 @@
...
+For the purpose of this demo, we will run a local MySQL Docker container:
+
+```
+docker run --rm -d --name atlas-demo -p 3306:3306 -e MYSQL_ROOT_PASSWORD=pass -e MYSQL_DATABASE=default mysql:8
+```
+
+Now, let's apply our schema to the database:
+
+```
+atlas schema apply -u mysql://root:pass@localhost:3306/default -f schema.hcl
```
Comment (a8m) [inline doc/website/blog/2023-10-19-atlas-v-0-15.mdx:265]:
```suggestion
atlas schema apply -u mysql://root:pass@localhost:3306/default --to file://schema.hcl
```

**Review by a8m — APPROVED** [review]

**Review by a8m — APPROVED** [review]

**Pull request merged**